### PR TITLE
Add --profile build argument

### PR DESCRIPTION
**Pull request by Johennes**
Fixes: #190

### CODE REVIEW

File: crates/ubrn_cli/src/ios.rs
Comment (jhugman) [inline crates/ubrn_cli/src/ios.rs:217]: Nit: extract a local variable
```suggestion
        let profile = self.common_args.profile();
        if profile != "debug" {
            cmd.args(["--profile", profile]);
```

File: crates/ubrn_cli/src/android.rs
Comment (jhugman) [inline crates/ubrn_cli/src/android.rs:240]: Nit: use local variable
```suggestion
        if profile != "debug" {
            cmd.args(["--profile", profile]);
```

File: crates/ubrn_cli/src/android.rs
Comment (jhugman) [inline crates/ubrn_cli/src/android.rs:235]: Nit: extract a local variable
```suggestion
        let profile = self.common_args.profile();
        if profile != "release" {
```

File: xtask/src/run/rust_crate.rs
Comment (jhugman) [inline xtask/src/run/rust_crate.rs:27]: This can be a bit fiddly sometimes, but I think adding `conflicts_with_all` to either this attribute or `release` should make `--profile` and `--release` mutually exclusive.

```suggestion
    #[clap(long, requires = "crate_dir", conflicts_with_all = ["release"])]
```

**Review by jhugman — APPROVED** [review]
@Johennes You're just so quick! Thank you so much.

I've approved this with nits. I don't think they're big enough to wait for another review cycle, but I would like these fixed before landing.

Also: would you mind adding an entry to the `CHANGELOG.md`? I think this qualifies as "✨ What's New"

File: xtask/src/run/rust_crate.rs
Comment (Johennes) [reply xtask/src/run/rust_crate.rs:27]: Oh, nice! I wasn't aware of this feature. 🤯

**Comment by Johennes** [comment]
> Also: would you mind adding an entry to the CHANGELOG.md? I think this qualifies as "✨ What's New"

Oh, I completely forgot that we have a changelog now. 🤦‍♂️

**Pull request merged**
